src/OBJ/Assembler.elm

```diff
@@ -223,7 +219,22 @@ addFaceToMesh f mesh ({ vs, vts, vns, currentIndex } as state) =
 
         _ ->
             -- TODO: lift this error into a Result type
-            Debug.crash "mixed face types in the model!"
+            Debug.log "ERROR: mixed face types in the model!" <|
```

Comment:
If I recall correctly, we can't use any of the `Debug` functions in libraries or production builds.
So I think we probably just have to ignore the error.
Or lift this function into a `Result` type, but that will be more work.

Reply:
Ah, yes :disappointed: 

>Or lift this function into a Result type, but that will be more work.

You're Not wrong there! I've been trying it for a couple of hours, the ripple effects are pretty huge (there are several places this needs fixing, resulting in a lot of functions that are lifted to `Result`. 

The problem is very few things have type annotations so it's inferred type errors only, and they're hard to read. So I tried adding them everywhere, but it's hard to see which errors are which now...
Does this look right?

```elm
type alias CompileState =
    {
    groups: Groups
    , currentGroupName: String
    , currentMaterialName : String
    , currentMesh : Maybe MeshT
    , currentGroup : Dict String Mesh
    , vs : Array Vec3
    , vts : Array Vec2
    , vns : Array Vec3
    , currentIndex : Int
    , knownVertexTextures : Dict Int3 Int
    , knownVertexTexturesTangents : Dict Int Int3
    , knownVertex : Dict Int2 Int
    , config : Settings
    }

emptyCompileState : Settings -> CompilerState
emptyCompileState config = ...
```

Reply:
Yeah, unfortunately I also gave up back then.. 
And I'm sorry, I can't really help you more here, I'm not familiar with the code anymore.

Reply:
OK. Perhaps I'll open that as a separate issue and leave it for now